prot.c

```diff
@@ -459,13 +460,7 @@ process_queue()
     int64 now = nanoseconds();
 
     while ((j = next_awaited_job(now))) {
-        heapremove(&j->tube->ready, j->heap_index);
-        ready_ct--;
-        if (j->r.pri < URGENT_THRESHOLD) {
-            global_stat.urgent_ct--;
-            j->tube->stat.urgent_ct--;
-        }
-
+        j = remove_ready_job(j);
```

Comment:
j could be NULL here and should be checked for that.

Reply:
Thanks for review, I added precaution check.